Language feature - Tooltip

Wolfsblvt opened the pull request:
Addint the language strings as tooltip for collapsible buttons.

Wolfsblvt committed:
Adding tooltip for collapse buttons

Wolfsblvt committed:
Fix formatting in html template

Wolfsblvt committed:
Changing order of parameter to alphabetically

iMattPro commented on event/listener.php:
```diff
@@ -82,6 +87,9 @@ public function init_collapsible_categories()
 	public function show_collapsible_categories($event)
 	{
+		// Add collapsible-categories language file
+		$this->user->add_lang_ext('phpbb/collapsiblecategories', 'collapsiblecategories');
```
This should not be here. This event takes place inside a loop and gets repeated multiple times.

Also, since our other event above is being removed in another PR, and because this should be available for other extensions to possibly use, this lang stuff should be handled using the core.user_setup model:
https://github.com/phpbb-extensions/boardrules/blob/master/event/listener.php#L80-L87

iMattPro commented on event/listener.php:
```diff
@@ -32,13 +35,15 @@ class listener implements EventSubscriberInterface
 	 * @access public
 	 */
-	public function __construct(\phpbb\controller\helper $helper, \phpbb\collapsiblecategories\operator\operator_interface $operator, \phpbb\template\template $template)
+	public function __construct(\phpbb\user $user, \phpbb\controller\helper $helper, \phpbb\collapsiblecategories\operator\operator_interface $operator, \phpbb\template\template $template)
```
Should be alphabetical, but not needed if lang is loaded in the core.user_setup

iMattPro commented on language/en/collapsiblecategories.php:
```diff
@@ -0,0 +1,42 @@
+//
+
+$lang = array_merge($lang, array(
+	'COLLAPSIBLE_CATEGORIES_TOOLTIP'		=> 'Click to show / hide category',
```
Should be _TITLE instead of _TOOLTIP I think since it's for the title attribute

Wolfsblvt committed:
Fix listener language + tests

Changed language from user function to core.user_setup event. Added
tests for new method.

Wolfsblvt committed:
JS call .prev() just once

Wolfsblvt committed:
Improve test method

iMattPro commented on event/listener.php:
```diff
@@ -73,6 +74,23 @@ public function init_collapsible_categories()
 	}
 
 	/**
+	 * Load common language files during user setup
```
For better organization, the functions should appear in the class in the same order that they are called in getSubscribedEvents.

iMattPro replied:
But it's OK for now since I am removing the other function above anyways.

Wolfsblvt commented:
I wasn't sure how you order them.
Is there any specific order, or just a group them (if many events) and then functions in the same order as in the events function?

iMattPro commented:
Alphabetically order the events.
